python/python/glide/sync/glide_client.py

```diff
@@ -48,8 +49,19 @@ def create(cls, config: BaseClientConfiguration) -> Self:
         self._init_ffi()
         self.config = config
         self._is_closed = False
-        conn_req = config._create_a_protobuf_conn_request(
-            cluster_mode=type(config) is GlideClusterClientConfiguration
+
+        os.register_at_fork(after_in_child=self._reset_client_connection)
```

Comment:
so why do we need both _create_new_core_client and _reset_client_connection? can't we just register _create_new_core_client?